guide/src/python_from_rust.md

```diff
@@ -230,3 +230,51 @@ def leaky_relu(x, slope=0.01):
 
 [`Python::run`]: https://docs.rs/pyo3/latest/pyo3/struct.Python.html#method.run
 [`py_run!`]: https://docs.rs/pyo3/latest/pyo3/macro.py_run.html
+
+## Need to use a context manager from Rust?
+
+Use context managers by directly invoking `__enter__` and `__exit__`.
+
+```rust
+use pyo3::prelude::*;
+use pyo3::types::PyModule;
+
+fn main() {   
+    Python::with_gil(|py| {
+        let custom_manager = PyModule::from_code(py, r#"
+class House(object):
+    def __init__(self, address):
+        self.address = address
+    def __enter__(self):
+        print(f"Welcome to {self.address}!")
+    def __exit__(self, type, value, traceback):
+        if type:
+            print(f"Sorry you had {type} trouble at {self.address}")
+        else:
+            print(f"Thank you for visiting {self.address}, come again soon!")
+
+        "#, "house.py", "house").unwrap();
+        
+        let house = custom_manager.call_function1("House", ("123 Main Street",)).unwrap();
+
+        house.call_function0("__enter__").unwrap();
```

Comment:
Confusingly, this one _should_ be call `call_method0` - the new name `call_function` is just for `&PyModule`. But now the confusion makes me think this was a mistake.

Sorry 😬 

```suggestion
        house.call_method0("__enter__").unwrap();
```